src/Neo/SmartContract/Native/NeoToken.cs

```diff
@@ -327,13 +327,34 @@ public BigInteger UnclaimedGas(DataCache snapshot, UInt160 account, uint end)
             return CalculateBonus(snapshot, state, end);
         }
 
+        [ContractMethod(Hardfork.HF_Echidna, RequiredCallFlags = CallFlags.States | CallFlags.AllowCall | CallFlags.AllowNotify)]
```

Comment:
Regarding `CallFlags.AllowCall` flag: we already discussed that with @roman-khimov in https://github.com/nspcc-dev/neo-go/pull/3700#discussion_r1856329113, but I'd probably like to see others' opinion.

This flag is added here to reflect an externall call to `GAS.Burn`. However, I'd say that it's not a common pattern for our native contracts and we need to be aligned in behaviour throughout the whole set of natives. For example, Neo's `getCandidates` handler makes a call to native Policy to check if candidate is blocked, however `getCandidates` requires only `CallFlags.States`. In general, native contract use `AllowCall` flag only if fully-qualified contract call should be made (with a proper new context loaded to VM and etc). Currently, there are only two places where `AllowCall` is required: Oracle's `finish` method that calls user-defined contract and native NEP-17 `transfer` method that calls `onNEP17Payment` of the receiver's contract.

I'd say that for now we should remove `CallFlags.AllowCall` from the set of required flags for this method, because `await GAS.Burn` is not a fully-qualified external contract call, it doesn't push new context to VM, doesn't push arguments on stack, doesn't check call flags. My primary concern is to have a unified behaviour and to be aligned with the rest of native methods, so we either fix similar cases for all other native contracts or remove `CallFlags.AllowCall` from this method.

Reply:
does this mean CallFlags.AllowCall is not taking effect for native at all?

Reply:
Nothing restricts natives really, they can break many limits (change the state of some non-native contract? easy).

Reply:
@AnnaShaleva, I think your reference to other methods is relevant here after all. To be more consistent we can remove `CallFlags.AllowCall` for this set.